portal/src/util/misc.ts

```diff
@@ -15,3 +15,24 @@ export function setFieldIfChanged(
     map[field] = value;
   }
 }
+
+export function clearEmptyObject(obj: Record<string, any>): void {
+  for (const key in obj) {
```

Comment:
When using `for ... in`, we should check call to check if the enumerated key is own properties.

```
for (const key in someObject) {
  // Use this form to call this function to ensure we are calling the function we intend to.
  // The reason is that hasOwnProperty may be overridden by some malicious code. 
  if (!Object.prototype.hasOwnProperty.call(someObject, key) { continue; }
}
```